### DISCUSSION TITLE
Change check in interval as config option

### DISCUSSION THREAD

**Discussion by CodemanCodes** [quartznet/quartznet#1331] [Ideas] [answered]
I apologize if this is already possible or not possible at all.

We are using clustering with sql server and I see the check in interval is hardcoded at 7500 milliseconds.

For our use case we only need to run jobs during a specific time span each day so each instance checking in with the DB so often is wasteful, unless I'm missing something (entirely possible).

Would a configuration option to change that interval be possible?

**Comment by lahma** [comment, marked as answer]
What you observe as hard-coded value is actually a configurable one. Quartz can inject values based on writable properties and in this casse the default can be overridden with a property:

```
quartz.jobStore.clusterCheckinInterval = 20000
```

Or with [DI configuration](https://www.quartz-scheduler.net/documentation/quartz-3.x/packages/microsoft-di-integration.html#di-aware-job-factories):

```csharp
q.UsePersistentStore(s =>
{
     // your code configuring
    s.UseClustering(c =>
    {
        c.CheckinInterval = TimeSpan.FromSeconds(20);
    });
});
```

**Reply by CodemanCodes** [reply]
Thanks! I had a feeling I was missing something, I guess diving further into the code or documentation would've helped.

Thank you for taking the time to reply.
